File: php/src/Behavioral/Visitor/AbstractTitleInfo.php
<?


abstract class AbstractTitleInfo
{
	private $titleName;

	abstract function accept( TitleBlurbVisitor &$tbv );
	
	function __construct( $titleName )
	{
		$this->setTitleName( $titleName );
	}
	
	function setTitleName($title)
	{
		$this->titleName = $title;
	}
	function getTitleName()
	{
		return $this->titleName;
	}
}

File: csharp/src/Behavioral/Visitor/AbstractTitleInfo.cs
namespace Behavioral.Visitor {


//AbstractTitleInfo.java - the abstract Visitee

public abstract class AbstractTitleInfo {  
   private string titleName;    
   public void setTitleName(string titleNameIn) {
       this.titleName = titleNameIn;
   }
   public string getTitleName() {
       return this.titleName;
   }
    
   public abstract void accept(TitleBlurbVisitor titleBlurbVisitor);
}

}
